STAGING - 22.0.0.3 GA Blog (#2079)
OpenLiberty/blogs: posts/2022-03-15-ga-release-post.adoc

seo-description: Open Liberty 22.0.0.3 allows SQL Operations to be retried in transaction recovery logs. This release also includes several significant bug fixes. In addition, we have published a brand new Open Liberty guide on containerizing microservices with Podman!
blog_description: Open Liberty 22.0.0.3 allows SQL Operations to be retried in transaction recovery logs. This release also includes several significant bug fixes. In addition, we have published a brand new Open Liberty guide on containerizing microservices with Podman!

open-graph-image: https://openliberty.io/img/twitter_card.jpg
---
= Retrying SQL operations in transaction recovery logs and more in Open Liberty 22.0.0.3
Ryan Storey <https://github.com/ryan-storey>
:imagesdir: /
:url-prefix:
:url-about: /
//Blank line here is necessary before starting the body of the post.

Open Liberty 22.0.0.3 allows SQL Operations to be retried in transaction recovery logs. This release also includes several significant bug fixes. In addition, we have published a brand new Open Liberty guide on containerizing microservices with Podman!
In link:{url-about}[Open Liberty] 22.0.0.3:

> * <<sql, Allow SQL Operations to be retried in transaction recovery logs>>

Reviewer: ```suggestion
* <<sql, Allow SQL Operations to be retried in transaction recovery logs>>
* <<bugs, Notable bug fixes>>
```

A new attribute is available for the `transaction` configuration element in the Liberty `server.xml` file that enables these transaction log SQL operations to be retried. It is named `enableLogRetries` and should be set to `true` to enable retries. For example,

[source,xml]
----
<transaction
dataSourceRef="tranlogDataSource"
recoverOnStartup="true"
waitForRecovery="false"
heuristicRetryInterval="10"
enableLogRetries="true"
/>
----

By default the new functionality is disabled.

For more information on the storing or Transaction Recovery Logs in an `RDBMS`, please refer to link:{url-prefix}/docs/latest/transaction-service.html#log[Transaction log management] doc.

[#bugs]
== Notable bugs fixed in this release

We’ve spent some time fixing bugs. The following sections describe just some of the issues resolved in this release. If you’re interested, here’s the  link:https://github.com/OpenLiberty/open-liberty/issues?q=label%3Arelease%3A22003+label%3A%22release+bug%22+[full list of bugs fixed in 22.0.0.3].

* link:https://github.com/OpenLiberty/open-liberty/issues/20165[jsonpContainer-2.0 and jsonbContainer-2.0 features incorrectly use default providers]
+

> The `jsonpContainer-2.0` and `jsonbContainer-2.0` features incorrectly used default providers. These features have been updated to include the BELLS feature and remove Liberty's default implementations of JSON-P and JSON-B. This allows for a user to include their own third party library for these JSON technologies. Additionally this fix updates the `JSON` container `FAT` tests to run against the 2.0 json container features (Jakarta EE 9) to verify this fixed functionality.

Reviewer: ```suggestion
The `jsonpContainer-2.0` and `jsonbContainer-2.0` features incorrectly used default providers. These features have been updated to include the BELLS feature and remove Liberty's default implementations of JSON-P and JSON-B. This allows for a user to include their own third party library for these JSON technologies. 
```

* link:https://github.com/OpenLiberty/open-liberty/issues/12050[@RolesAllowed rejects unauthenticated users when they mapped to an allowed (EVERYONE) role]
+

> Previously, the special-subject `EVERYONE` did not play well with the `@RolesAllowed` annotation on `JAX-RS` resources. When the special-subject `EVERYONE` is bound to a `security-role`, unauthenticated users should be considered to be in that role and should have access to the endpoints allowed for that role. The `@RolesAllowed` annotation would reject unauthenticated users when they were mapped to an allowed (`EVERYONE`) role. This issue has since been resolved.

Reviewer: ```suggestion
Previously, the special-subject `EVERYONE` did not work correctly with the `@RolesAllowed` annotation on `JAX-RS` resources. When the special-subject `EVERYONE` is bound to a `security-role`, unauthenticated users should be considered to be in that role and should have access to the endpoints allowed for that role. The `@RolesAllowed` annotation would reject unauthenticated users when they were mapped to an allowed (`EVERYONE`) role. This issue has since been resolved.
```

* link:https://github.com/OpenLiberty/open-liberty/issues/19673[JWT access token inbound propagation fails when a JWT sent as segments starts with "Bearer"]
+
In 22.0.0.2, `JWT` access token inbound propagation failed when a `JWT` sent as segments starts with `Bearer`. Liberty has support to accept multiple headers from `WebSeal` that include portions of a `JWT` access token. The first header, `Authorization-segments`, for example, indicates how many "n" segments follow. The rest of the headers `Authorization-1` through `Authorization-n` contain the `JWT` access token. When `Authorization-1` started with `Bearer`, the access token could not be parsed correctly for inbound propagation. If JWT segments are sent and the first header, `Authorization-segments-1`, starts with `Bearer`, it should be processed correctly as when a JWT is sent using the `Authorization: Bearer`.

* link:https://github.com/OpenLiberty/open-liberty/issues/19960[OpenID Connect: Double URL Encoded State Parameter in Redirect location]
+

> Previously, when receiving a state parameter from the relying party (RP) it appears Liberty returned the state double encoded. The issue appears to occur when processing the liberty page to authorize the RP:

Reviewer 2: ```suggestion
Previously, when receiving a state parameter from the relying party (RP) it appears Liberty returned the state double encoded. The issue appears to occur when processing the Liberty page to authorize the RP:
```

+
[source]
----
https://www.oauth-login.com/oidc/endpoint/rp/authorize
Payload state: yyyyyyyyyyyyyyyyy
----
+
Which results in a response code `302`: and a redirect location of:
+

> `https://oauth-openshift.apps.ocp4.example.com/oauth2callback/openid?code=xxxxxxxx&state=zzzzzzzzzzzzzzzzzz`

Reviewer 2: ```suggestion
`https://rp.example.com/oauth2callback/openid?code=xxxxxxxx&state=zzzzzzzzzzzzzzzzzz`
```
